"""corrige problema com a relação de serviço e agenamento 

Revision ID: 93d8485816de
Revises: 843f0c56d2a9
Create Date: 2025-05-03 03:16:12.220839

"""
from alembic import op
import sqlalchemy as sa
from sqlalchemy.dialects import postgresql

# revision identifiers, used by Alembic.
revision = '93d8485816de'
down_revision = '843f0c56d2a9'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('agendamento_servico',
    sa.Column('agendamento_id', sa.UUID(), nullable=False),
    sa.Column('servico_id', sa.UUID(), nullable=False),
    sa.ForeignKeyConstraint(['agendamento_id'], ['agendamentos.id'], ),
    sa.ForeignKeyConstraint(['servico_id'], ['servicos.id'], ),
    sa.PrimaryKeyConstraint('agendamento_id', 'servico_id')
    )
    with op.batch_alter_table('agendamentos', schema=None) as batch_op:
        batch_op.drop_column('servico_ids')

    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('agendamentos', schema=None) as batch_op:
        batch_op.add_column(sa.Column('servico_ids', postgresql.ARRAY(sa.UUID()), autoincrement=False, nullable=False))

    op.drop_table('agendamento_servico')
    # ### end Alembic commands ###
